### ISSUE TITLE
auth: invalid LOC normalization

### ISSUE THREAD

**Issue by peterthomassen** [PowerDNS/pdns#10558]
- Program: Authoritative
 - Issue type: Bug report

### Short description
For certain valid LOC record contents, the API returns an error: `Not in expected format`.

### Environment
 - Operating system: Ubuntu
 - Software version: 4.4.1-1pdns.bionic
 - Software source: PowerDNS repository

### Steps to reproduce
1. Provision a LOC record with content `49 18 0.000 N 10 35 0.000 E 409.00m` using the API

### Expected behaviour
record accepted

### Actual behaviour
The API returns an error: `Not in expected format (parsed as '49 17 60.000 N 10 35 0.000 E 409.00m 1.00m 10000.00m 10.00m'`

### Other information
@Habbie stated on IRC that that's a bug.

**Comment by Habbie** [comment]
Yep! 18 minutes and 0 seconds should be represented as 18 and 0, not 17 and 60. The 60 is out of range.

**Comment by cjsoftuk** [comment]
Turns out that this exists in both directions.  Today alone I have experienced the error:

`'57 9 0.000 N 2 7 9.998 W 0.00m 0.00m 0.00m 0.00m': Not in expected format (parsed as '57 8 60.000 N 2 7 9.998 W 0.00m 0.00m 0.00m 0.00m')`

I have also experienced `'51 11 60.000 N 2 12 5.080 W 0.00m 0.00m 0.00m 0.00m': Not in expected format (parsed as '51 12 0.000 N 2 12 5.080 W 0.00m 0.00m 0.00m 0.00m'`

@Habbie thinks this is a rounding error somewhere in the madness that is LOC record parsing.

**Comment by omoerbeek** [comment]
https://github.com/omoerbeek/pdns/tree/loc-lat-long seems to work and is much easier on the eye. Will PR once I have reviewed the code one more time.

**Comment by omoerbeek** [comment]
Reading https://www.rfc-editor.org/rfc/rfc1876, it must be noted that the LAT and LONG both allow for skipping parts. The REST validation using our own idea of what's canonical is strict and unforgiving but also arbitrary. E.g. the origin has four representations with respect to hemisphere (vary N/S and E/W), but only one will be accepted. And that's not even using the possibility to leave minutes and seconds out if they are zero.
